somaticseq/annotate_caller.py

```diff
@@ -0,0 +1,338 @@
+    if variant_id in varscan_variants:
+
+        varscan_variant_i = varscan_variants[ variant_id ]
+        varscan_classification = 1 if varscan_variant_i.get_info_value('SOMATIC') else 0
+
+        # If ref_base, first_alt, and indel_length unknown, get it here:
+        #if not ref_base:         ref_base = varscan_variant_i.refbase
+        #if not first_alt:        first_alt = varscan_variant_i.altbase
+        #if indel_length == None: indel_length = len(first_alt) - len(ref_base)
+        
+    else:
+        varscan_classification = 0
+
+    return varscan_classification
+
+
+
+
+def JSM(variant_id, jsm_variants):
+    
+    if variant_id in jsm_variants:
+        
+        jsm_variant_i = jsm_variants[ variant_id ]
+        jointsnvmix2_classification = 1
+        aaab = float( jsm_variant_i.get_info_value('AAAB') )
+        aabb = float( jsm_variant_i.get_info_value('AABB') )
+        jointsnvmix2_p = 1 - aaab - aabb
+        score_jointsnvmix2 = genome.p2phred(jointsnvmix2_p, max_phred=50)
+
+        # If ref_base, first_alt, and indel_length unknown, get it here:
+        #if not ref_base:         ref_base = jsm_variant_i.refbase
+        #if not first_alt:        first_alt = jsm_variant_i.altbase
+        #if indel_length == None: indel_length = len(first_alt) - len(ref_base)
+        
+    else:
+        jointsnvmix2_classification = 0
+        score_jointsnvmix2 = nan
+
+    return jointsnvmix2_classification, score_jointsnvmix2
+
+
+
+
+def SomaticSniper(variant_id, sniper_variants):
+    if variant_id in sniper_variants:
+        
+        sniper_variant_i = sniper_variants[ variant_id ]
+        sniper_classification = 1 if sniper_variant_i.get_sample_value('SS', idxT) == '2' else 0
+        if sniper_classification == 1:
+            score_somaticsniper = sniper_variant_i.get_sample_value('SSC', idxT)
+            score_somaticsniper = int(score_somaticsniper) if score_somaticsniper else nan
+        else:
+            score_somaticsniper = nan
+
+        # If ref_base, first_alt, and indel_length unknown, get it here:
+        #if not ref_base:         ref_base = sniper_variant_i.refbase
+        #if not first_alt:        first_alt = sniper_variant_i.altbase
+        #if indel_length == None: indel_length = len(first_alt) - len(ref_base)
+            
+    else:
+        sniper_classification = 0
+        score_somaticsniper = nan
+
+    return sniper_classification, score_somaticsniper
+
+
+
+
+def VarDict(variant_id, vardict_variants):
+    
+    if variant_id in vardict_variants:
+        
+        vardict_variant_i = vardict_variants[ variant_id ]
+        
+        if (vardict_variant_i.filters == 'PASS') and ('Somatic' in vardict_variant_i.info):
+            vardict_classification = 1
+        
+        elif 'Somatic' in vardict_variant_i.info:
+            vardict_filters = vardict_variant_i.filters.split(';')
+            
+            disqualifying_filters = ('d7' in vardict_filters or 'd5' in vardict_filters) or \
```

Comment:
Should mostly be a set intersection. Something like `DISQUAL_FILTERS = set(['d7', 'd5', ...]) & set(vardict_filters)`

Reply:
Things gets a bit more complicated toward the end of those "DISQUAL_FILTERS," when I look for INFO given certain filters.